proposals/0032-inline-stylesheets.md

```diff
@@ -0,0 +1,89 @@
+
+- Start Date: 2023-04-15
+- Reference Issues: [#556](https://github.com/withastro/roadmap/issues/556)
+- Implementation PR: [withastro/astro#6659](https://github.com/withastro/astro/pull/6659)
+
+# Summary
+
+Provide a configuration to control inlining behavior of styles authored or imported in astro modules.
+
+# Example
+
+```ts
+export default defineConfig({
+    build: {
+        // all styles necessary are sent in external stylsheets, default; maintains current behavior
+        inlineStylesheets: "never"
+    }
+})
+```
+```ts
+export default defineConfig({
+    build: {
+        // stylesheets smaller than `ViteConfig.build.assetsInlineLimit` (default: 4kb) are inlined
```

Comment:
Relying on the external tool has another drawback that we should highlight: if `vite` decides to change its defaults, we are forced to create a **major** release because it changes **astro's defaults**, unless we explicitly say that this API is subject to Vite's behaviour and Astro's doesn't intend to respect it.

Reply:
We are relying not on vite, but on vite subconfig in Astro config, the [defaults](https://github.com/withastro/astro/blob/main/packages/astro/src/core/config/schema.ts#L36) for which [we decide](https://github.com/withastro/astro/blob/2d8dad1a/packages/astro/src/core/build/plugins/plugin-css.ts#L222).

Reply:
Sorry, that wasn't clear! (I don't know the internals of that part)

Reply:
You are right, that isn't clear. At one point, I realized it might sound like 4kb is vite's default being used, but I forgot about it soon.

Editing to clarify.